beacon_node/lighthouse_network/src/peer_manager/mod.rs

```diff
@@ -1285,6 +1285,30 @@ impl<TSpec: EthSpec> PeerManager<TSpec> {
             );
         }
     }
+
+    /// Checks if the peer is banned, if so
+    /// disconnects and queues `PeerManagement::Banned` so that discovert also bans it.
+    fn check_ban_peer(&mut self, peer_id: PeerId) -> Option<BanResult> {
+        match self.ban_status(&peer_id) {
+            Some(cause @ BanResult::BadScore) => {
+                // This is a faulty state
+                error!(self.log, "Connection attempt with a banned peer. Re-banning"; "peer_id" => %peer_id);
+                // Disconnect the peer.
+                self.goodbye_peer(&peer_id, GoodbyeReason::Banned, ReportSource::PeerManager);
+                // Re-ban the peer to prevent repeated errors.
+                self.events.push(PeerManagerEvent::Banned(peer_id, vec![]));
```

Comment:
`goodbye_peer` already goes through a set of heuristics via:
- `PeerManager::report_peer` 
- which in turn calls `PeerManager::handle_score_action`  
- and subsequently `PeerManager::handle_ban_operation` which for `BanOperation::TemporaryBan`'s calls `self.events.push(PeerManagerEvent::Banned(*peer_id, vec![]));`
calling here `self.events.push(PeerManagerEvent::Banned(*peer_id, vec![]));` doesn't result in a redundant event returned to the `Swarm`?

Reply:
I think that's somewhat intentional since this is doing some reactive bug correction

Reply:
In principle yes. We should never reach this state. The idea was that if everything works as expected, banned peers should not be able to form connections with us. 

There have been some weird bugs in the past, and potentially if the swarm itself or libp2p forgets about which peers get banned (for whatever reason) where we end up in this situation. The situation is such that a peer that we think is banned has managed to connect to us. 

We emit an error, because this shouldn't happen and we tell the swarm again to re-ban it, if for some reason the first ban didn't work or if the swarm forgot or there was some bug. 

So it is redundant, but the redundancy has helped in the past when other parts of the code has broken. Rather than letting a banned peer constantly re-connect, we manage to re-ban it in the case of a fault.

Reply:
Thanks for explaining Age.
since we are moving from the checks from `FromSwarm::ConnectionEstablished` to `handle_established_inbound_connection` and `handle_established_outbound_connection` banning the peers won't impede them from attempting to connect us  again, and these functions will be called again. 
Should we still push the `PeerManagerEvent::Banned`? If so, should we then also push on `BanResult::BannedIp`?